lib/nss.c

```diff
@@ -0,0 +1,150 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <dlfcn.h>
+#include <stdbool.h>
+#include <string.h>
+#include <strings.h>
+#include <ctype.h>
+#include "prototypes.h"
+#include "../libsubid/subid.h"
+
+#define NSSWITCH "/etc/nsswitch.conf"
+
+// NSS plugin handling for subids
+// If nsswitch has a line like
+//    subid: sssd
+// then sssd will be consulted for subids.  Unlike normal NSS dbs,
+// only one db is supported at a time.  That's open to debate, but
+// the subids are a pretty limited resource, and local files seem
+// bound to step on any other allocations leading to insecure
+// conditions.
+static bool nss_initialized;
+
+static struct subid_nss_ops *subid_nss;
+
+struct subid_nss_ops *get_subid_nss_handle() {
```

Comment:
Did you consider to call `nss_init()` here?
Looks like it could save a line of code in every `lib/subordinateio.c ` function without impacting readability...
(Just IMO)

Reply:
No I hadn't considered it.  That's a good idea.